src/groupBy.js

```diff
@@ -45,10 +42,7 @@ var append = require('./append');
  *      //   'F': [{name: 'Eddy', score: 58}]
  *      // }
  */
-module.exports = _curry2(_dispatchable('groupBy', _xgroupBy, function groupBy(fn, list) {
-  return _reduce(function(acc, elt) {
-    var key = fn(elt);
-    acc[key] = append(elt, acc[key] || (acc[key] = []));
-    return acc;
-  }, {}, list);
-}));
+module.exports = _dispatchable('groupBy', _xgroupBy, reduceBy(function(acc, item) {
+  acc ? acc.push(item) : acc = [item]
```

Comment:
So apart that curry thing (sorry for misleading you earlier!), I prefer a proper ``if () { ...; return ...; } return ...`` here.